src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs

```diff
@@ -65,13 +65,13 @@ public static class MessageDrivenSubscriptionsConfigExtensions
 
             ThrowOnAddress(publisherEndpoint);
 
-            routingSettings.Settings.GetOrCreate<ConfiguredPublishers>().Add((publishers, knownMessageTypes) =>
+            routingSettings.Settings.GetOrCreate<ConfiguredPublishers>().Add((publishers, conventions) =>
             {
-                foreach (var knownMessageType in knownMessageTypes)
+                foreach (var type in assembly.GetTypes())
```

Comment:
not sure. A comment would be required to prevent if someone wants to add missing exception handling, which I don't really see as a huge danger?